### DISCUSSION TITLE
go fmt lr-files

### DISCUSSION THREAD

**Discussion by mariuskimmina** [mondoohq/cnquery#1247] [Q&A] [answered]
I noticed that running `go fmt ./...` in the root directory of the `cnquery` project would result in a lot of changes in .lr.go files.  Are these lr.go files auto generated how have they been manually created? Also what does `lr` actually mean here?

**Comment by chris-rock** [comment, marked as answer]
Hi @mariuskimmina. Excellent question. Historically our MQL resource packs had the code name "Lumi" which was derived from the German word "Mond". `lr` files are simply lumi resource files. We plan to rename the file extension to `mqlr` since they represent the MQL resources.

**Comment by chris-rock** [comment]
Based on that schema we generate the go boilerplate with the `lr` compiler in https://github.com/mondoohq/cnquery/blob/main/Makefile#L78-L128. When you add a resource, you just need to run `make cnspec/generate` and the lr.go files will be generated. Developer instructions are https://github.com/mondoohq/cnquery/blob/main/docs/development.md. If there is anything else missing, let us know.
